### DISCUSSION TITLE
Dynamically changing workspace output base on number of monitors connected

### DISCUSSION THREAD

**Discussion by baoke344** [i3/i3#5511] [Q&A]
Hi guys!

How can I change the output of working space base on number of connected

Normally, at my home, I have 2 monitor connected. But at my work office, I have only one 
So, How can I config my i3 base on that ?
Thanks

**Comment by lkuty** [comment]
Maybe you could play with `xrandr` and `i3-msg` using a script. Or use two configuration files and switch between them to handle different workspace numbering scheme. I guess that you want to put some workspaces on a display and the rest of them on the other one e.g. `i3-msg "[workspace=2] move workspace to output eDP-1"`. Thus, you could have two scripts :  one for home, the other for work. Both scripts are filled with `i3-msg` `move workspace` commands to distribute the workspaces on the outputs.

**Comment by orestisfl** [comment]
The question is not really clear enough regarding what you want to do. A starting point however is that you can assign workspaces to multiple outputs, so you can have a preferred order of workspace assignment to outputs: https://i3wm.org/docs/userguide.html#workspace_screen

E.g. `workspace 5 output VGA1 LVDS1` means that workspace 5 will be assigned to VGA1 if it exists and LVDS1 if not

Also, the `primary` and `nonprimary` keywords are supported in many commands/config directives.
